src/openrct2/entity/Guest.cpp

```diff
@@ -1656,32 +1656,38 @@ bool Guest::DecideAndBuyItem(Ride& ride, const ShopItem shopItem, money64 price)
     GiveItem(shopItem);
     const auto hasRandomShopColour = ride.HasLifecycleFlag(RIDE_LIFECYCLE_RANDOM_SHOP_COLOURS);
 
-    if (shopItem == ShopItem::TShirt)
-        TshirtColour = hasRandomShopColour ? ScenarioRandMax(kColourNumNormal) : ride.track_colour[0].main;
-
-    if (shopItem == ShopItem::Hat)
-        HatColour = hasRandomShopColour ? ScenarioRandMax(kColourNumNormal) : ride.track_colour[0].main;
-
-    if (shopItem == ShopItem::Balloon)
-        BalloonColour = hasRandomShopColour ? ScenarioRandMax(kColourNumNormal) : ride.track_colour[0].main;
-
-    if (shopItem == ShopItem::Umbrella)
-        UmbrellaColour = hasRandomShopColour ? ScenarioRandMax(kColourNumNormal) : ride.track_colour[0].main;
-
-    if (shopItem == ShopItem::Map)
-        ResetPathfindGoal();
-
-    if (shopItem == ShopItem::Photo)
-        Photo1RideRef = ride.id;
-
-    if (shopItem == ShopItem::Photo2)
-        Photo2RideRef = ride.id;
-
-    if (shopItem == ShopItem::Photo3)
-        Photo3RideRef = ride.id;
-
-    if (shopItem == ShopItem::Photo4)
-        Photo4RideRef = ride.id;
+    switch (shopItem)
+    {
+        case ShopItem::TShirt:
+            TshirtColour = hasRandomShopColour ? ScenarioRandMax(kColourNumNormal) : ride.track_colour[0].main;
+            break;
+        case ShopItem::Hat:
+            HatColour = hasRandomShopColour ? ScenarioRandMax(kColourNumNormal) : ride.track_colour[0].main;
+            break;
+        case ShopItem::Balloon:
+            BalloonColour = hasRandomShopColour ? ScenarioRandMax(kColourNumNormal) : ride.track_colour[0].main;
+            break;
+        case ShopItem::Umbrella:
+            UmbrellaColour = hasRandomShopColour ? ScenarioRandMax(kColourNumNormal) : ride.track_colour[0].main;
+            break;
+        case ShopItem::Map:
+            ResetPathfindGoal();
+            break;
+        case ShopItem::Photo:
+            Photo1RideRef = ride.id;
+            break;
+        case ShopItem::Photo2:
+            Photo2RideRef = ride.id;
+            break;
+        case ShopItem::Photo3:
+            Photo3RideRef = ride.id;
+            break;
+        case ShopItem::Photo4:
+            Photo4RideRef = ride.id;
+            break;
+        default:
+            break; // Fallback
```

Comment:
Comments should tell us something. This just states the obvious.

```suggestion
            break;
```